apiserver/facades/client/spaces/shims.go

```diff
@@ -103,27 +100,6 @@ func (s *stateShim) SubnetByCIDR(cidr string) (networkingcommon.BackingSubnet, e
 	return networkingcommon.NewSubnetShim(result), nil
 }
 
-func (s *stateShim) RenameSpace(settingsChanges settings.ItemChanges, constraints map[string]constraints.Value, fromSpaceName, toName string) error {
-	err := s.State.
-		RenameSpace(fromSpaceName, toName, settingsChanges, constraints)
-	if err != nil {
-		return errors.Trace(err)
-	}
-	return nil
-}
-
-func (s *stateShim) ConstraintsBySpace(spaceName string) (map[string]constraints.Value, error) {
-	result, err := s.State.ConstraintsBySpaceName(spaceName)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	return result, nil
-}
-
-func (s *stateShim) ControllerConfig() (controller.Config, error) {
-	result, err := s.State.ControllerConfig()
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
-	return result, nil
+func (s *stateShim) ApplyOperation(op state.ModelOperation) error {
```

Comment:
You shouldn't need to shim this method. A `*state.State` supplies it directly.

Reply:
I thought we did not want that and that's why we did not add it in the first place and used a Backing instead.

If that's not the case I can add one to the struct 
// API provides the spaces API facade for version 6.
type API struct {
        *state.State <-?
	backing   Backing
	resources facade.Resources
	auth      facade.Authorizer
	context   context.ProviderCallContext

	check     BlockChecker
	opFactory OpFactory
}

Reply:
No need for that, just delete it and you will see it builds fine.
`*state.State` is embedded in `stateShim`. 

The only reason you need to implement methods is when transforming between state package types and your local shims.